client/reader/index.js

```diff
@@ -29,7 +30,11 @@ function forceTeamA8C( context, next ) {
 
 export default function() {
 	if ( config.isEnabled( 'reader' ) ) {
-		page( '/', preloadReaderBundle, initAbTests, updateLastRoute, sidebar, following );
+		if ( config.isEnabled( 'restore-last-location' ) ) {
+			page( '/', sessionRestore, preloadReaderBundle, initAbTests, updateLastRoute, sidebar, following );
```

Comment:
> Would it be a goo idea to skip preloading the reader bundle at this point?

From what I can see, if `sessionRestore` decides to restore a previous session, it will never call `next`, thus breaking this middleware chain and not preloading the reader.